### ISSUE TITLE
Naming convention for static mut is not enforced

### ISSUE THREAD

**Issue by matklad** [rust-lang/rust#37145]
The following program does not report any naming convention warnings:

```rust
#![allow(dead_code)]

static mut snake_case: u32 = 0;
static mut camelCase: u32 = 0;
static mut SCREAMING_SNAKE_CASE: u32 = 0;
static mut CapitalizedCamelCase: u32 = 0;

fn main() {}
```

Looks like only `SCREAMING_SNAKE_CASE` should be allowed? If I remove `mut`s, I get the warnings. Or have we already deprecated static muts (pre rfc: https://internals.rust-lang.org/t/pre-rfc-remove-static-mut/1437/20) ?

**Comment by sfackler** [comment]
Static muts are not deprecated. This looks like it's just an oversight in those lints. The convention should be identical to normal statics.

**Comment by matklad** [comment]
Ok, I'll try to submit a PR then.
